app/sidekiq/terms_of_use/sign_up_service_updater_job.rb

```diff
@@ -48,23 +48,42 @@ def log_updated_icn
       end
     end
 
+    def map_client
+      @map_client ||= MAP::SignUp::Service.new
+    end
+
+    def map_status
+      @map_status ||= map_client.status(icn: mpi_profile.icn)
+    end
+
+    def agreement_unchanged?
+      if terms_of_use_agreement.declined? != map_status[:opt_out] ||
+         terms_of_use_agreement.accepted? != map_status[:agreement_signed]
+        return false
+      end
```

Comment:
this seems a bit complex. what if `map_status` returned `'accepted'` or `'declined'` and then we just compare `tems_of_use_agreement.response` to `map_status`?

something like:

```ruby
def agreement_unchanged?
  return false unless terms_of_use_agreement.response == map_status

  true
end
```

Reply:
Ultimately I'll leave it up to Emily to choose her preferred way, but I think there's no way getting around just a bit of awkwardness around the logic here (due to the strange way that Sign Up Service has made their response). 

This solution makes the `agreement_unchanged?` function simpler, but it basically moves the conditional inside the `map_status` function, in addition to encoding the `enum` string values inside `map_values` instead of taking advantage of enum functions

Reply:
I think it's okay to leave the conditional logic inside `agreement_unchanged?` in the case `map_status` may be used for in another function down the road. But I'm happy to change it for a simpler solution as well. I'll leave it as it is for now if that's alright.

Reply:
I ultimately think there should be a MAP response object where all this logic lives but thats probably out of scope for this. Something to consider if we end up doing more with these responses